### ISSUE TITLE
Sum over Pow incorrectly gives nan in some cases

### ISSUE THREAD

**Issue by albertz** [sympy/sympy#18524]
Let
```
i = Symbol("i", integer=True)
n = Symbol("n", integer=True, nonnegative=True)
x = Symbol("x", real=True)
y = Symbol("y", real=True)
```

Then `Sum(Pow(x, i) * Pow(1 - y, n - i), (i, 1, n - 1)).subs(n, 3).subs(x, 1.).subs(y, 1.).doit()` gives `nan`.
If I change `subs(x, 1.)` to `subs(x, 1)`, it does not give `nan`.
I'm not exactly sure why. Probably some incorrect simplification or rewriting?

**Comment by Jonty16117** [comment]
ok, so let's iterate the problems in your equation one by one:
1. You didn't mention the value of **"n"** if you are trying to get a constant result
2. **"1."** in **"subs(x, 1.)"** and **"subs(y, 1.)"** is not valid since it is neither a float nor an integer, so here is this mistake.
3. If you are not trying to find the constant answer to this problem, let's say you want to simplify the result in terms of **"n"** (since you didn't define it), then it still gives **"nan"** because of the **2 problem** mentioned above.

Code:
```
>>> Sum(Pow(x, i) * Pow(1 - y, n - i), (i, 1, n - 1)).subs(n, 3).subs(x, 1.).subs(y, 1.).subs(i, 0).doit()
nan
>>> Sum(Pow(x, i) * Pow(1 - y, n - i), (i, 1, n - 1)).subs(n, 3).subs(x, 1).subs(y, 1).subs(i, 0).doit()
0
>>> Sum(Pow(x, i) * Pow(1 - y, n - i), (i, 1, n - 1)).subs(n, 3).subs(x, 1).subs(y, 0).subs(i, 1).doit()
2
```

**Comment by albertz** [comment]
1. But I have `subs(n, 3)` in there. Or what do you mean?
2. I don't understand. `1.` is a float. What exactly is wrong?
3. I want to find the constant answer here in this example.

**Comment by oscarbenjamin** [comment]
Here is a simpler reproducer:
```julia
In [5]: s = Sum(0.0**(3-n)*2**n, (n, 1, 2))                                                                                                  

In [6]: s                                                                                                                                      
Out[6]: 
  2              
 ___             
 ╲               
  ╲      3 - n  n
  ╱   0.0     ⋅2 
 ╱               
 ‾‾‾             
n = 1            

In [7]: s.doit()                                                                                                                               
Out[7]: nan
```

**Comment by oscarbenjamin** [comment]
I'm guessing that there's a `0/0` somewhere in the calculation.

**Comment by sylee957** [comment]
```python3
from sympy import *

i = Symbol("i", integer=True)
n = Symbol("n", integer=True, nonnegative=True)
x = Symbol("x", real=True)
y = Symbol("y", real=True)

Sum(Pow(x, i) * Pow(1-y, n-i), (i, 1, n-1)).doit()
```
The closed-form solution gives 
![image](https://user-images.githubusercontent.com/34944973/73595527-1122f480-455d-11ea-9fae-db3929ada09f.png)
Even though 
`Sum(Pow(x, i) * Pow(1-y, n-i), (i, 1, n-1)).subs(n, 3).doit()`gives a better result
![image](https://user-images.githubusercontent.com/34944973/73595543-59421700-455d-11ea-8ec3-83a2666ffa42.png)
that doesn't require the division

NaN can probably be from some division by zero, but there can be something wrong going in the summation to give the result which needs `y` not equal to 1.

**Comment by albertz** [comment]
How do you get this closed form solution? By using `simplify`? And the closed form solution is wrong for the case `y == 1`, right?

**Comment by Jonty16117** [comment]
> 1. But I have `subs(n, 3)` in there. Or what do you mean?
> 2. I don't understand. `1.` is a float. What exactly is wrong?
> 3. I want to find the constant answer here in this example.

Yes, you are right, I didn't read the equation properly. Sorry, my mistake :(
